core/src/modules/ops/record.ts

```diff
@@ -831,12 +831,16 @@ export namespace RecordOps {
     }
 
     const recordProperties = await RecordProperty.findAll({
-      where: { unique: true, rawValue: rawValues },
+      where: {
+        propertyId: uniqueProperties.map((p) => p.id),
```

Comment:
This addition will include only those existing records for the properties in this model/source now